README.Rmd

```diff
@@ -21,360 +21,112 @@ knitr::opts_chunk$set(
-    -   [Table and Field information](#table-and-field-information)
-    -   [Reading](#reading)
-    -   [Writing](#writing)
-    -   [Querying](#querying)
--   [Benchmarks](#benchmarks)
-    -   [Reading](#reading-1)
-    -   [Writing](#writing-1)
-
-## Installation
-For Unix and MacOS ODBC drivers should be compiled against
-[unixODBC](https://www.unixodbc.org/). Drivers compiled against [iODBC](https://www.iodbc.org/) _may_ also
-work, but are not fully supported.
-
-After installation of the driver manager and driver, you will have to
-register the driver in a [odbcinst.ini](#dsn-configuration-files) file for it
-to appear in `odbc::odbcListDrivers()`.
-
-odbc and its dependencies use C++11 features. Therefore you need [gcc
-4.8](https://gcc.gnu.org/), [clang 3.3](https://clang.llvm.org) or [Rtools
-3.3](https://CRAN.R-project.org/bin/windows/Rtools/) or later.
-
-### Windows
-Windows is bundled with ODBC libraries. However, drivers for each database
-need to be installed separately. Windows ODBC drivers typically include an
-installer that needs to be run and will install the driver to the proper
-locations.
-
-### MacOS
-[homebrew](https://brew.sh/) can be used to easily install database drivers on MacOS.
-
-#### UnixODBC - Required for all databases
-```shell
-# Install the unixODBC library
-brew install unixodbc
-```
-
-#### Common DB drivers
-```shell
-# Microsoft SQL Server ODBC Drivers (Recommended)
-brew tap microsoft/mssql-release https://github.com/Microsoft/homebrew-mssql-release
-brew update
-brew install msodbcsql17 mssql-tools
+The goal of the odbc package is to provide a [DBI](https://dbi.r-dbi.org/)-compliant interface to [ODBC](https://learn.microsoft.com/en-us/sql/odbc/microsoft-open-database-connectivity-odbc) drivers. This makes it easy to connect databases such as [SQL Server](https://www.microsoft.com/en-us/sql-server/), [Oracle](https://www.oracle.com/database), [Databricks](https://www.databricks.com/), and [Snowflake](https://www.snowflake.com/).
 
-# SQL Server ODBC Drivers (Free TDS)
-brew install freetds
+The odbc package is an alternative to
+[RODBC](https://cran.r-project.org/package=RODBC) and [RODBCDBI](https://cran.r-project.org/package=RODBCDBI) packages, and is typically much faster. See `vignette("benchmarks")` to learn more.
 
-# PostgreSQL ODBC ODBC Drivers
-brew install psqlodbc
+## Overview
 
-# MySQL ODBC Drivers (and database)
-brew install mysql
+The odbc package is one piece of the R interface to databases with support for ODBC:
 
-# SQLite ODBC Drivers
-brew install sqliteodbc
+```{r whole-game, eval = TRUE, echo = FALSE}
+#| fig-alt: > 
+#|   A diagram containing four boxes with arrows linking each pointing left to 
+#|   right. The boxes read, in order, "R interface," "driver manager," 
+#|   "ODBC driver," and "DBMS." The left-most box, R interface, contains 
+#|   three smaller components, labeled "dbplyr," "DBI," and "odbc."
+knitr::include_graphics("man/figures/whole-game.png")
```

Comment:
I think it might make sense to flip this diagram vertically so that reading from left-to-right follows the paragraphs below.

Reply:
Agreed that it'd be nice if the order that the image labels would be read matches the order of the paragraphs. Flip vertically as in make the diagram vertical or flip along the y axis?

A:
![Screenshot 2023-12-11 at 3 03 29 PM](https://github.com/r-dbi/odbc/assets/35748691/6d82e72f-a922-4786-ab2d-c684a165e5c3)

B: 
![Screenshot 2023-12-11 at 3 05 36 PM](https://github.com/r-dbi/odbc/assets/35748691/eae4c54f-a412-40ba-b5b9-543332f0509c)

The arrows here point in the same "direction" but end up Right to Left which feels weird, but I guess the flow of Database to R Interface vs R Interface to Database is somewhat arbitrary (following the direction of the data to the interface vs the the query to the database). Maybe B but reverse the direction of the arrows?

Reply:
I was thinking B and reverse direction of the arrows

Reply:
[ca13b35](https://github.com/r-dbi/odbc/pull/652/commits/ca13b350887f47db96fc7f2ce9df4ca0ae5ec56e)!